src/umpire/Umpire.cpp

```diff
@@ -277,6 +284,15 @@ MPI_Comm get_communicator_for_allocator(Allocator a, MPI_Comm comm)
 
   return c;
 }
+
+void cleanup_cached_communicators()
+{
+  std::map<int, MPI_Comm>& comm = get_cached_communicators();
+
+  for (auto c : comm) {
+    MPI_Comm_free(&c.second);
+  }
```

Comment:
I think you may want to call clear on this map after freeing all the communicators.

Reply:
Done!